plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcPageSourceFactory.java

```diff
@@ -437,6 +450,37 @@ else if (transaction.isUpdate()) {
         }
     }
 
+    private static void validateOrcAcidVersion(Optional<AcidInfo> acidInfo, boolean originalFile, OrcReader reader)
+    {
+        if (!originalFile && acidInfo.isPresent() && !acidInfo.get().isOrcAcidVersionValidated()) {
+            int writerId = reader.getFooter().getWriterId();
+            if (writerId == TRINO_WRITER_ID || writerId == PRESTO_WRITER_ID) {
+                // file written by Trino. We are good.
+                return;
+            }
+
+            Optional<Integer> hiveAcidVersion = getHiveAcidVersion(reader);
+            if (hiveAcidVersion.isEmpty() || hiveAcidVersion.get() < 2) {
+                throw new TrinoException(NOT_SUPPORTED, "Hive transactional tables are supported since Hive 3.0. " +
+                        "If you have upgraded from an older version of Hive, make sure a major compaction has been run at least once after the upgrade.");
```

Comment:
Include that we expected `hive.acid.version` in the file and what was the value we found.
this can help people troubleshoot problems

also, if this is thrown, will the problematic file path be added to the exc msg later on?

Reply:
It is logged. But is not part of exception message retuned from query. I will add it there.

Reply:
Now error message is:
```
20210413_143740_00006_vrwxw failed: Hive transactional tables are supported since Hive 3.0. Expected `hive.acid.version` in ORC metadata in hdfs://hadoop-master:9000/user/hive/warehouse/t/delta_0000001_0000001_0000/bucket_00002 to be >=2 but was 1. If you have upgraded from an older version of Hive, make sure a major compaction has been run at least once after the upgrade.
```